### ISSUE TITLE
Runs locally but deploying to azure gives "Error uploading archive (InternalServerError)"

### ISSUE THREAD

**Issue by rramani** [Azure/azure-functions-core-tools#1305]
I am using a python based azure function that takes the forecasting automated ML sample file (.pkl), loads it with the dependencies and does a scoring. I have an appservice plan for linux and then deploying the functionapp to it. Getting the error during deployment but starts locally fine and runs correctly.

(.functionenv) Rama Mac:blog rama$ func azure functionapp publish forecastamlfunctionapp --build-native-deps
Your functionapp is using a custom image DOCKER|appsvc/azure-functions-runtime.
Assuming that the image contains the correct framework.

Microsoft (R) Build Engine version 15.9.20+g88f5fadfbe for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 48.22 ms for /Users/rama/azure_functions/blog/extensions.csproj.
  extensions -> /Users/rama/azure_functions/blog/bin/extensions.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.27

Getting site publishing info...
Assuming --nozip (do not run from package) for publishing to Linux dedicated plan.
Creating archive for current directory...
Directory .python_packages already in sync with requirements.txt. Skipping restoring dependencies...
Uploading 851.78 MB [#############################################################################]
Creating archive for current directory...
Directory .python_packages already in sync with requirements.txt. Skipping restoring dependencies...
Uploading 851.78 MB [#############################################################################]
Creating archive for current directory...
Directory .python_packages already in sync with requirements.txt. Skipping restoring dependencies...
Uploading 851.78 MB [#############################################################################]
Error uploading archive (InternalServerError).

requirements.txt
(.functionenv) Rama Mac:blog rama$ cat requirements.txt 
azure-functions==1.0.0b3
azure-functions-worker==1.0.0b3
grpcio==1.14.2
grpcio-tools==1.14.2
protobuf==3.6.1
six==1.12.0
numpy
azureml-sdk
sklearn

(.functionenv) Rama Mac:blog rama$ cat host.json 
{
    "version":  "2.0"
}

I have a 'HttpTrigger' folder: 

(.functionenv) Rama Mac:HttpTrigger rama$ ls
__pycache__		function.json		main.py			main.py_url_params
forecastmodel.pkl	host.json		main.py_static_data

(.functionenv) Rama Mac:HttpTrigger rama$ func host start
your worker runtime is not set. As of 2.0.1-beta.26 a worker runtime setting is required.
Please run `func settings add FUNCTIONS_WORKER_RUNTIME <option>` or add FUNCTIONS_WORKER_RUNTIME to your local.settings.json
Available options: dotnet, node, python, powershell

                  %%%%%%
                 %%%%%%
            @   %%%%%%    @
          @@   %%%%%%      @@
       @@@    %%%%%%%%%%%    @@@
     @@      %%%%%%%%%%        @@
       @@         %%%%       @@
         @@      %%%       @@
           @@    %%      @@
                %%
                %

Azure Functions Core Tools (2.7.1158 Commit hash: f2d2a2816e038165826c7409c6d10c0527e8955b)
Function Runtime Version: 2.0.12438.0
SKipping 'FUNCTIONS_CORETOOLS_ENVIRONMENT' because value is null
[5/14/19 6:21:07 PM] Starting Rpc Initialization Service.
[5/14/19 6:21:07 PM] Initializing RpcServer
[5/14/19 6:21:07 PM] Building host: startup suppressed:False, configuration suppressed: False
[5/14/19 6:21:08 PM] Initializing Host.
[5/14/19 6:21:08 PM] Host initialization: ConsecutiveErrors=0, StartupCount=1
[5/14/19 6:21:08 PM] LoggerFilterOptions
[5/14/19 6:21:08 PM] {
[5/14/19 6:21:08 PM]   "MinLevel": "None",
[5/14/19 6:21:08 PM]   "Rules": [
[5/14/19 6:21:08 PM]     {
[5/14/19 6:21:08 PM]       "ProviderName": null,
[5/14/19 6:21:08 PM]       "CategoryName": null,
[5/14/19 6:21:08 PM]       "LogLevel": null,
[5/14/19 6:21:08 PM]       "Filter": "<AddFilter>b__0"
[5/14/19 6:21:08 PM]     },
[5/14/19 6:21:08 PM]     {
[5/14/19 6:21:08 PM]       "ProviderName": "Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics.SystemLoggerProvider",
[5/14/19 6:21:08 PM]       "CategoryName": null,
[5/14/19 6:21:08 PM]       "LogLevel": "None",
[5/14/19 6:21:08 PM]       "Filter": null
[5/14/19 6:21:08 PM]     },
[5/14/19 6:21:08 PM]     {
[5/14/19 6:21:08 PM]       "ProviderName": "Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics.SystemLoggerProvider",
[5/14/19 6:21:08 PM]       "CategoryName": null,
[5/14/19 6:21:08 PM]       "LogLevel": null,
[5/14/19 6:21:08 PM]       "Filter": "<AddFilter>b__0"
[5/14/19 6:21:08 PM]     }
[5/14/19 6:21:08 PM]   ]
[5/14/19 6:21:08 PM] }
[5/14/19 6:21:08 PM] FunctionResultAggregatorOptions
[5/14/19 6:21:08 PM] {
[5/14/19 6:21:08 PM]   "BatchSize": 1000,
[5/14/19 6:21:08 PM]   "FlushTimeout": "00:00:30",
[5/14/19 6:21:08 PM]   "IsEnabled": true
[5/14/19 6:21:08 PM] }
[5/14/19 6:21:08 PM] SingletonOptions
[5/14/19 6:21:08 PM] {
[5/14/19 6:21:08 PM]   "LockPeriod": "00:00:15",
[5/14/19 6:21:08 PM]   "ListenerLockPeriod": "00:00:15",
[5/14/19 6:21:08 PM]   "LockAcquisitionTimeout": "10675199.02:48:05.4775807",
[5/14/19 6:21:08 PM]   "LockAcquisitionPollingInterval": "00:00:05",
[5/14/19 6:21:08 PM]   "ListenerLockRecoveryPollingInterval": "00:01:00"
[5/14/19 6:21:08 PM] }
[5/14/19 6:21:08 PM] Starting JobHost
[5/14/19 6:21:08 PM] Starting Host (HostId=ramamac-185699412, InstanceId=d2f3afcc-454a-46b9-b9cc-245b49d25dc7, Version=2.0.12438.0, ProcessId=73250, AppDomainId=1, InDebugMode=False, InDiagnosticMode=False, FunctionsExtensionVersion=)
[5/14/19 6:21:08 PM] Loading functions metadata
[5/14/19 6:21:08 PM] 0 functions loaded
[5/14/19 6:21:08 PM] WorkerRuntime: (null). Will shutdown other standby channels
[5/14/19 6:21:08 PM] Generating 0 job function(s)
[5/14/19 6:21:08 PM] No job functions found. Try making your job classes and methods public. If you're using binding extensions (e.g. Azure Storage, ServiceBus, Timers, etc.) make sure you've called the registration method for the extension(s) in your startup code (e.g. builder.AddAzureStorage(), builder.AddServiceBus(), builder.AddTimers(), etc.).
[5/14/19 6:21:08 PM] Host initialized (94ms)
[5/14/19 6:21:08 PM] Host started (104ms)
[5/14/19 6:21:08 PM] Job host started
Hosting environment: Production
Content root path: /Users/rama/azure_functions/blog/HttpTrigger
Now listening on: http://0.0.0.0:7071
Application started. Press Ctrl+C to shut down.
[5/14/19 6:21:13 PM] Host lock lease acquired by instance ID '0000000000000000000000002C1DEACA'.
[5/14/19 6:21:48 PM] Executed HTTP request: {
[5/14/19 6:21:48 PM]   "requestId": "f24b4799-8017-491f-b2a3-72f4009cd4f3",
[5/14/19 6:21:48 PM]   "method": "GET",
[5/14/19 6:21:48 PM]   "uri": "/api/HttpTrigger",
[5/14/19 6:21:48 PM]   "identities": [],
[5/14/19 6:21:48 PM]   "status": 404,
[5/14/19 6:21:48 PM]   "duration": 414
[5/14/19 6:21:48 PM] }

Ramasubramanians-MacBook-Pro:blog rama$ curl -X GET 'http://127.0.0.1:7071/api/HttpTrigger?clientId=ramamac'

**Comment by jtlz2** [comment]
Isn't the clue in the log..? :\

```
your worker runtime is not set. As of 2.0.1-beta.26 a worker runtime setting is required.
Please run func settings add FUNCTIONS_WORKER_RUNTIME <option> or add FUNCTIONS_WORKER_RUNTIME to your local.settings.json
Available options: dotnet, node, python, powershell
```

**Comment by rramani** [comment]
Setting the variable does not resolve the issue. Getting 'BadGateway' error

**Comment by ankitkumarr** [comment]
Update: the problem might be caused by the size of the package you are publishing. This issue should be resolved when https://github.com/Azure/azure-functions-core-tools/issues/1069 is closed. The main work has been done, just waiting on changes to roll out. I will keep this thread updated.

**Comment by jeroenroodnat** [comment]
Is there any progress on deployment? We are running into same issue that container package is not deploying whereas running it on our own docker server has no issues.

**Comment by ankitkumarr** [comment]
Apologies, I missed the ping here. This issue should be resolved. 

Unlikely, but if you do see any such errors, please open a new issue as we may need to look at those case by case.
